src/FamilyHubs.ServiceDirectory.Web/StartupExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿using FamilyHubs.ServiceDirectory.Infrastructure.Services.PostcodesIo.Extensions;
+using FamilyHubs.ServiceDirectory.Infrastructure.Services.ServiceDirectory.Extensions;
+using Microsoft.ApplicationInsights.Extensibility;
+using Serilog;
+using Serilog.Events;
+
+namespace FamilyHubs.ServiceDirectory.Web;
+
+public static class StartupExtensions
+{
+    public static void ConfigureHost(this WebApplicationBuilder builder)
+    {
+        // ApplicationInsights
+        builder.Host.UseSerilog((_, services, loggerConfiguration) =>
+        {
+            var logLevelString = builder.Configuration["LogLevel"];
+
+            var parsed = Enum.TryParse<LogEventLevel>(logLevelString, out var logLevel);
+
+            loggerConfiguration.WriteTo.ApplicationInsights(
+                services.GetRequiredService<TelemetryConfiguration>(), 
+                TelemetryConverter.Traces, 
+                parsed ? logLevel : LogEventLevel.Warning);
+
+            loggerConfiguration.WriteTo.Console(
+                parsed ? logLevel : LogEventLevel.Warning);
+        });
+    }
+
+    public static void ConfigureServices(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddApplicationInsightsTelemetry();
+
+        // Add services to the container.
+        services.AddRazorPages();
+        services.AddPostcodesIoClient(configuration);
+        services.AddServiceDirectoryClient(configuration);
+    }
+
+    public static IServiceProvider ConfigureWebApplication(this WebApplication app)
+    {
+        app.UseSerilogRequestLogging();
+
+        // Configure the HTTP request pipeline.
+        if (!app.Environment.IsDevelopment())
+        {
+            app.UseExceptionHandler("/Error");
+            // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+            app.UseHsts();
+        }
+
+        app.UseHttpsRedirection();
```

Comment:
We don't want this at the moment (but it might come back later).

Reply:
added #if use_https, sorry I must have missed it